core/src/main/java/io/questdb/cairo/O3PartitionJob.java

```diff
@@ -1380,6 +1461,8 @@ private static void mergeRowGroup(
                     long dstFixSize = mergeRowCount * ColumnType.sizeOf(columnType);
                     final long dstFixMemAddr = Unsafe.malloc(dstFixSize, MemoryTag.NATIVE_O3);
 
+                    // TODO(eugene): can be null in case of column top
```

Comment:
nit: can we add an assertion or a check with a thrown exception that the pointer is valid for now? It's better than failing implicitly.